tests/one_taskset.rs

```diff
@@ -41,14 +42,31 @@ fn test_single_taskset() {
     config.users = Some(2);
     config.hatch_rate = 4;
     config.status_codes = true;
+
+    let mut paths_and_weights = vec![(INDEX_PATH, 9), (ABOUT_PATH, 3)];
+    let mut taskset = GooseTaskSet::new("LoadTest");
+
+    while let Some(item) = paths_and_weights.pop() {
+        let path = item.0;
+        let weight = item.1;
+
+        let closure: GooseTaskFunction = Arc::new(move |user| {
+            Box::pin(async move {
+                let _goose = user.get(path).await?;
+                Ok(())
+            })
+        });
+
+        let mut task = GooseTask::new(closure).set_weight(weight).unwrap();
+        // @todo Cannot use taskset.register_task as that won't work in a loop.
+        task.tasks_index = taskset.tasks.len();
+        taskset.tasks.push(task);
+    }
```

Comment:
And here is a first programmatic test 🎉

Reply:
Can we split this out into its own test function, perhaps `test_single_taskset_closure` or `test_closure`, instead of adding this to an existing test? As all tests within a single file run in parallel, it helps to see what's actually failing when tests fail. Plus, integration tests can be useful demonstrations of how to do something: for someone needing to use closures we can direct them to the closure-specific test. (And perhaps add a second test that demonstrates/tests closures together with passing in functions?)

Reply:
done